src/targets/browser/browserProcess.ts

```diff
@@ -0,0 +1,90 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+import * as stream from 'stream';
+
+type stdioType = [
+  NodeJS.WritableStream, // stdin
+  NodeJS.ReadableStream, // stdout
+  NodeJS.ReadableStream, // stderr
+  NodeJS.ReadableStream | NodeJS.WritableStream | null | undefined, // extra, no modification
+  NodeJS.ReadableStream | NodeJS.WritableStream | null | undefined, // extra, no modification
+];
+
+export interface IBrowserProcess {
+  readonly stdio: stdioType;
+  readonly pid: number;
+  readonly stderr: NodeJS.ReadableStream;
+  readonly stdout: NodeJS.ReadableStream;
+  removeListener(eventName: 'exit', onExit: (code: number) => void): void;
+  removeListener(eventName: 'error', onError: (error: Error) => void): void;
+  on(eventName: string, listener: Function): void;
+}
+
+export class BrowserProcessByPid implements IBrowserProcess {
+  public constructor(public readonly pid: number) {}
+
+  // TODO: Figure out how to use a WeakSet here so we don't have a memory leak (At the moment we don't use it because it can't be iterated)
+  private static instancesToPoll = new Set<BrowserProcessByPid>();
+
+  private static pollInterval: NodeJS.Timeout | null = null;
+  private exitListeners = new Set<Function>();
+
+  public get stdio(): stdioType {
```

Comment:
Assuming you added this in the interface to avoid type errors creating the `PipeTransport` down below, I wonder if `launch()` could be restructured to avoid needing to to this?

Reply:
I can't think of any easy way, if you come up with something please let me know